skimage/transform/registration.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from scipy.optimize import basinhopping, minimize
+
+from .pyramids import pyramid_reduce
+from ._warps import warp, SimilarityTransform
+
+__all__ = ['register', 'p_to_matrix', 'matrix_to_p']
+
+
+def _gaussian_pyramid(image, levels=6):
+    """
+    Creates a gaussian pyramid for an image
+    
+    Parameters
+    ----------
+    image : (M, N) ndarray
+        Input image with is used as a base for the pyramid.
+    levels : int, optional (default 6)
+        Int equals to the amount of times the image is blurred and reduced
+    
+    Returns
+    -------
+    pyramid: array
+        Ordered array of images of size level starting with the most blurred
+        
+    """
+    pyramid = levels*[None]
+    pyramid[-1] = image
+
+    for level in range(levels-2, -1, -1):
+        image = pyramid_reduce(image, sigma=2/3)
+        pyramid[level] = (image)
+
+    return pyramid
+
+
+def _mse(img1, img2):
+    """
+    Finds the error between the overlapping area of two images
+    
+    Parameters
+    ----------
+    img1 : (M, N) ndarray
+        Input image used for reference
+    img2 : (M, N) ndarray
+        Input image which is compared to img1
+    
+    Returns
+    -------
+    err: int
+        Error in the form of the squared difference between pixels
+    """
+    dim0 = min(img1.shape[0],img2.shape[0])
```

Comment:
PEP8, space after comma

Reply:
Actually, this whole function needs to be updated to support n-dimensional images:

```
limits = tuple(slice(None, min(s, t)) for s, t in zip(img1.shape, img2.shape))
return np.sum((img1[limits] - img2[limits])**2)
```

Reply:
@leGIT-bot do you want this or shall I do it